### PR TITLE
add more default values

### PR DESCRIPTION
**Pull request by reeflog**
* title default value is the project name
* description is the project name if no description is provided
* files are taken from artifacts of msbuild plugin and the target
folder from target framework in csproj

### CODE REVIEW

File: src/main/groovy/com/ullink/NuGetSpec.groovy [hunk 1]
```diff
@@ -65,24 +61,54 @@ class NuGetSpec extends Exec {
...
             root.appendNode { metadata defaultValues }
         } else {
             root.metadata.appendNode defaultValues
         }
+
+        if (msbuildTaskExists) {
+            def defaultFiles = {}
+            project.msbuild.mainProject.dotnetArtifacts.each {
+                artifact ->
+                    def fwkFolder = fwkVersions.find {
```
Comment (gluck) [inline src/main/groovy/com/ullink/NuGetSpec.groovy]: sure this whole fwkVersions map is necessary ? we're only considering `project.msbuild.mainProject` here, so we could get its target fwk version with `project.msbuild.mainProject.properties.TargetFrameworkVersion...` no ?

Comment (reeflog) [inline src/main/groovy/com/ullink/NuGetSpec.groovy]: I believe at the beginning I had in mind to handle multiple projects. It's not really needed and currently it doesn't work. If we want to do so I think we will need to do more work around here.
I've removed it.

**Comment by gluck** [comment]
:shipit:

File: src/main/groovy/com/ullink/NuGetSpec.groovy [hunk 2]
```diff
@@ -65,24 +61,44 @@ class NuGetSpec extends Exec {
...
+
+        if (msbuildTaskExists) {
+            def defaultFiles = {}
+            project.msbuild.mainProject.dotnetArtifacts.each {
+                artifact ->
+                    def fwkFolderVersion = project.msbuild.mainProject.properties.TargetFrameworkVersion.toString().replace('v', '').replace('.', '')
+                    defaultFiles <<= { file(src: artifact.toString(), target: 'lib/net' + fwkFolderVersion) }
+            }
+
+            appendAndCreateParentIfNeeded('files', defaultFiles)
```
Comment (gluck) [inline src/main/groovy/com/ullink/NuGetSpec.groovy]: I missed the fact that this provide no way to choose which files gets added (all output from msbuild is added, always, and you can't exclude it), you can notice the effect of ILRepack build:
```  <files>
    <file src="C:\workspace\git-perso\il-repack\build\tmp\repack\ILRepack.dll" target="lib/net40"/>
    <file src="C:\workspace\git-perso\il-repack\ILRepack\bin\Release\ILRepack.exe" target="lib/net40"/>
    <file src="C:\workspace\git-perso\il-repack\ILRepack\bin\Release\ILRepack.pdb" target="lib/net40"/>
  </files>```

Comment (reeflog) [inline src/main/groovy/com/ullink/NuGetSpec.groovy]: I see few options.
- If files are already provided we don't add them.
- We define a "msbuildArtifactsAsNuspecFiles" in the nuspec task (false by default). If true, we add them.
- The same as the first but we define a "forceMsbuildArtifactsAsNuspecFiles" for those that want them to be added anyway.

I think the first one make more sense.

Comment (gluck) [inline src/main/groovy/com/ullink/NuGetSpec.groovy]: :+1:

Comment (reeflog) [inline src/main/groovy/com/ullink/NuGetSpec.groovy]: See https://github.com/Ullink/gradle-nuget-plugin/pull/29

**Pull request merged**